tools-local/tasks/mobile.tasks/WasmAppBuilder/WasmAssemblySearcher.cs

```diff
@@ -0,0 +1,159 @@
+
+        var paths = new List<string>();
+        _assemblies = new SortedDictionary<string, Assembly>();
+
+        if (AssemblySearchPaths != null)
+        {
+            // Collect and load assemblies used by the app
+            foreach (var v in AssemblySearchPaths!)
+            {
+                var dir = v.ItemSpec;
+                if (!Directory.Exists(dir))
+                    throw new ArgumentException($"Directory '{dir}' doesn't exist or not a directory.");
+                paths.Add(dir);
+            }
+            _resolver = new Resolver(paths);
+            var mlc = new MetadataLoadContext(_resolver, "System.Private.CoreLib");
+
+            var mainAssembly = mlc.LoadFromAssemblyPath(MainAssembly);
+            Add(mlc, mainAssembly);
+
+            if (ExtraAssemblies != null)
+            {
+                foreach (var item in ExtraAssemblies)
+                {
+                    try
+                    {
+                        var refAssembly = mlc.LoadFromAssemblyPath(item.ItemSpec);
+                        Add(mlc, refAssembly);
+                    }
+                    catch (System.IO.FileLoadException)
+                    {
+                        if (!SkipMissingAssemblies)
+                            throw;
+                    }
+                }
+            }
+        }
+        else
+        {
+            string corelibPath = string.Empty;
+            string runtimeSourceDir = string.Empty;
+            foreach (var v in Assemblies!)
+            {
+                if (v.ItemSpec.EndsWith ("System.Private.CoreLib.dll"))
+                    corelibPath = Path.GetDirectoryName (v.ItemSpec)!;
+            }
+            runtimeSourceDir = corelibPath!;
+            _resolver = new Resolver(new List<string>() { corelibPath });
+            var mlc = new MetadataLoadContext(_resolver, "System.Private.CoreLib");
+
+            foreach (var v in Assemblies!)
+            {
+                var assembly = mlc.LoadFromAssemblyPath(v.ItemSpec);
+                Add(mlc, assembly);
+            }
+        }
+
+        ReferencedAssemblies = GetOutputReferencedAssemblies();
+
+        return true;
+    }
+
+    private ITaskItem[] GetOutputReferencedAssemblies()
+    {
+        int count = 0;
+        ITaskItem[] items = new TaskItem[_assemblies!.Count];
+
+        foreach (var asm in _assemblies)
+        {
+            items[count++] = new TaskItem(asm.Value.Location);
+        }
+
+        return items;
+    }
+
+    private void Add(MetadataLoadContext mlc, Assembly assembly)
+    {
+        if (_assemblies!.ContainsKey(assembly.GetName().Name!))
+            return;
+        _assemblies![assembly.GetName().Name!] = assembly;
+        foreach (var aname in assembly.GetReferencedAssemblies())
+        {
+            try
+            {
+                Assembly refAssembly = mlc.LoadFromAssemblyName(aname);
```

Comment:
This can also throw `BadImageFormatException`, and `FileLoadException`. This can do the same thing as the exception handler for `LoadFromAssemblyPath`.

Reply:
How does `BadImageFormatException` get thrown?

Reply:
Invalid assembly. Try with a simple text file.